src/GregClient/AuthProviders/IAuthProvider.cs

```diff
@@ -45,5 +45,11 @@ public interface IOAuth2AuthProvider : IAuthProvider
     public interface IOAuth2AccessTokenProvider
     {
         string GetAccessToken();
+    }
+
+    // Auth Provider Interface to get user account information.
+    public interface IAuthProviderUserAccount
+    {
+        public string UserId { get; }
```

Comment:
would it be valuable while creating this interface to add other members, like username and email?

Reply:
Username is exposed in IAuthProvider, do you think it will be better usability to add it to this interface also?
Any security concerns related to exposing the user email address?

Reply:
Discussed with the team and we have a task in future to add other info. Going ahead with just Userid for the 3.1 release.